### PR TITLE
Add SoftDelete(tombstone option) to ProtoStore

### PR DESCRIPTION
**Pull request by krishnamiriyala**

### CODE REVIEW

File: pkg/datastore/database.go [hunk 1]
```diff
@@ -315,12 +315,59 @@ func (db *relationalDb) InsertInTable(ctx context.Context, tableName string, rec
...
+Deletes a record from DB table tableName. (to be used after soft-delete for structs with 'DeletedAt` fields).
+*/
+func (db *relationalDb) DeleteInTable(ctx context.Context, tableName string, record Record) (rowsAffected int64, err error) {
+	tx, err := db.GetDBTransaction(ctx, tableName, record)
+	if err != nil {
+		return 0, err
+	}
+
+	// Execute query
+	tx.Unscoped().Delete(record)
```
Comment (jeyhun) [inline pkg/datastore/database.go:361]: Shouldn't you check if `Delete` failed (by checking if `tx.Error` is not nil) before committing? (I see that everywhere in the code you're not checking `tx.Error` before committing)

**Comment by jeyhun** [comment]
@krishnamiriyala, I've noticed that in the unit test `TestTransactions` in _pkg/datastore/datastore_test.go_, you are only registering `pb.Disk` with ProtoStore, but not registering `App`, `AppUser`, `Group` with DataStore. Could you register those too, since they are used in that unit test?

File: pkg/protostore/protostore.go [hunk 2]
```diff
@@ -20,6 +20,8 @@
 // around [pkg.datastore.DataStore] interface and is used specifically to persist Protobuf messages.
 // Just as with [pkg.datastore.DataStore], Protobuf messages can be persisted with revisioning and
 // multi-tenancy support along with `CreatedAt` and `UpdatedAt` timestamps.
+// Tombstone Delete or soft deletes are supported with `DeletedAt` struct field.
+// Use Delete to do remove any records that are soft deleted but still in database
```
Comment (jeyhun) [inline pkg/protostore/protostore.go:24]: You should change "Use Delete to **do** remove any records that are soft deleted but still in database" to "Use Delete to remove any records that are soft deleted but still in database"

File: pkg/datastore/database.go [hunk 3]
```diff
@@ -315,12 +315,59 @@ func (db *relationalDb) InsertInTable(ctx context.Context, tableName string, rec
...
+		db.logger.Errorln(err)
+		return 0, err
+	}
+	return tx.RowsAffected, nil
+}
+
+/*
+Deletes a record from DB table tableName. (to be used after soft-delete for structs with 'DeletedAt` fields).
+*/
+func (db *relationalDb) DeleteInTable(ctx context.Context, tableName string, record Record) (rowsAffected int64, err error) {
```
[the remark this answers is not in the document]
Comment (krishnamiriyala) [reply pkg/datastore/database.go:354]: done

File: pkg/datastore/database.go (hunk 1 above)
Comment (krishnamiriyala) [reply pkg/datastore/database.go:361]: done

File: pkg/protostore/protostore.go (hunk 2 above)
Comment (jeyhun) [reply pkg/protostore/protostore.go:24]: @krishnamiriyala, I think you forgot to push/amend a commit

**Comment by krishnamiriyala** [comment]
> @krishnamiriyala, I've noticed that in the unit test `TestTransactions` in _pkg/datastore/datastore_test.go_, you are only registering `pb.Disk` with ProtoStore, but not registering `App`, `AppUser`, `Group` with DataStore. Could you register those too, since they are used in that unit test?

done

File: pkg/protostore/protostore.go [hunk 4]
```diff
@@ -20,6 +20,8 @@
 // around [pkg.datastore.DataStore] interface and is used specifically to persist Protobuf messages.
 // Just as with [pkg.datastore.DataStore], Protobuf messages can be persisted with revisioning and
 // multi-tenancy support along with `CreatedAt` and `UpdatedAt` timestamps.
+// Tombstone Delete or soft deletes are supported with `DeletedAt` struct field.
+// Use Delete to remove any records that are soft deleted but still in database
```
Comment (krishnamiriyala) [inline pkg/protostore/protostore.go:24]: I don't see "do" here @jeyhun

**Review by jeyhun — APPROVED** [review]
We will need to make sure that when inserting, upserting, and updating records, tx.Error is checked before tx.Commit() is called

File: pkg/protostore/protostore.go (hunk 4 above)
Comment (jeyhun) [reply pkg/protostore/protostore.go:24]: Yeah, it's not here anymore. It was there when I last looked

**Pull request merged**